tests/iface_namingmode/test_iface_namingmode.py

```diff
@@ -714,9 +714,15 @@ def test_config_interface_speed(self, setup_config_mode, sample_intf, duthosts,
         cli_ns_option = sample_intf['cli_ns_option']
         asic_index = sample_intf['asic_index']
         duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+        # Get supported speeds for interface
+        supported_speeds = duthost.get_supported_speeds(interface)
+        # Remove native speed from supported speeds
+        supported_speeds.remove(native_speed)
+        # Set speed to configure
+        configure_speed = supported_speeds[0] if supported_speeds else native_speed
```

Comment:
'get_supported_speeds' can also return None. Please add a check before trying to use the var 'supported_speeds'

Reply:
Should also check if it only supports the native speed - supported_speeds has only a single entry

Reply:
> Should also check if it only supports the native speed - supported_speeds has only a single entry
configure_speed = supported_speeds[0] if supported_speeds else native_speed - this code checks if supported_speeds  has at least one entry after removing native speed, if not - configure_speed = native_speed

Reply:
> 'get_supported_speeds' can also return None. Please add a check before trying to use the var 'supported_speeds'

@neethajohn  thank you for pointing on this, added check if supported_speeds != None